src/actions/lever/LeverSPTActions.sol

```diff
@@ -470,10 +470,11 @@ contract LeverSPTActions is Lever20Actions, ICreditFlashBorrower, IERC3156FlashB
             params.subPTokenAmount
         );
 
-        // approve the Sense Finance Adapter to transfer `target` tokens
-        if (params.redeemParams.approveTarget != 0) {
-            IERC20(params.redeemParams.target).approve(params.redeemParams.adapter, targetAmount);
+        // approve the Sense Finance Adapter to transfer max `target` tokens
+        if (IERC20(params.redeemParams.target).allowance(address(this),params.redeemParams.adapter) == 0) {
+            IERC20(params.redeemParams.target).safeApprove(params.redeemParams.adapter, type(uint256).max);
```

Comment:
if we set from 0 to max one time only then it should work with just approve

Reply:
We use safeApprove because USDT does not implement the `approve` method in the ERC20 standard. 
ERC20 `approve` should return a boolean and the USDT implementation does not so we get an EVM revert even if the initial allowance is 0 and we set it to a new value.

![image](https://user-images.githubusercontent.com/6408302/214785294-564eeef2-dbe9-4af2-a964-56df65d0dca5.png)

Reply:
this expects a boolean return value which usdt doesnt do https://github.com/OpenZeppelin/openzeppelin-contracts/blob/69c8d1010e1ca53ce3fa7f6fddff96759aec87e1/contracts/token/ERC20/IERC20.sol#L66

Reply:
It`s a bit funny that in the comment they link the ERC20 interface but they don`t use it :).

Reply:
that doesn't explain why it would revert. we never check the return value of approve. and the return params are not part of the method sig hash of the approve call.

Reply:
Yes but the evm expects a return when calling via ERC20 cast so it will revert even if we don`t check for the return value.
You can take a look at this repo for the test: https://github.com/amarcu/USDTTest

And I made this update because our vault tests(js) were failing for the USDT Vault and this fixes the test.